Refactor Database Models

chialuka opened the pull request:
#### What does this PR do?
- Refactors models to include validations, associations and custom functions

#### Description of Task proposed in this pull request?
- add validations to models
- add custom finder functions to models
- add associations with sessions to models
- replace helper finder functions with models finders

#### How should this be manually tested (Quality Assurance)?
- Clone the repo and run `npm test`. When the tests pass, it is evidence that the refactoring works because the code uses the new database functions.

#### Any background context you want to add (Operations Impact)?
- The `User` models file was refactored

- The `checkExistingUser` helper file was totally removed

#### What I have learned working on this feature:
- I learned how to create validations for models

- I learned how to write custom database functions

- I learned how to write associations for related database models

dinobi reviewed (commented):
❤️.  @LukasChiama, can you quickly add tests for the user model and all the functions there in.

chialuka committed:
refactor-[models]: refactor database models
- add validations to models
- add custom finder functions to models
- add associations with sessions to models
- replace helper finder functions with
- include tests for model validations and functions

models finders

chialuka committed:
Merge branch 'develop' into refactor-user-model

chialuka commented:
> ❤️. @LukasChiama, can you quickly add tests for the user model and all the functions there in.

Thank you. It is done
